core/src/main/java/org/infinispan/transaction/impl/TransactionTable.java

```diff
@@ -508,9 +508,8 @@ protected final void recalculateMinTopologyIdIfNeeded(CacheTransaction removedTr
    public void onTopologyChange(TopologyChangedEvent<?, ?> tce) {
       // don't do anything if this cache is not clustered
       if (clustered) {
-         if (tce.isPre()) {
+         if (!tce.isPre()) {
```

Comment:
Do we want a test to confirm this?  Unfortunately I am not as familiar with this.